Update dnslink.md (#369)
ipfs/docs: content/guides/concepts/dnslink.md

**NOTE: The info below is a quick guide to DNSLink. For a more complete guide, including tutorials, usage examples and FAQs, check out [dnslink.io](http://dnslink.io/).**

## What is DNSLink?

> DNSLink uses [DNS TXT](https://en.wikipedia.org/wiki/TXT_record) records to map a domain name (like `ipfs.io`) to an IPFS address. Because you can edit your DNS records, you can use them to always point to the latest version of an object in IPFS (remember that an IPFS object’s address changes if you modify the object). Because DNSLink uses DNS records, the names it produces are also usually easy to type and read.

Author: ```suggestion
DNSLink uses [DNS TXT](https://en.wikipedia.org/wiki/TXT_record) records to map a domain name (like `ipfs.io`) to an IPFS address. Because you can edit your DNS records, you can use them to always point to the latest version of an object in IPFS (remember that an IPFS object’s address changes if you modify the object). Since DNSLink uses DNS records, you can assign names/paths/(sub)domains/whatever that are easy to type, read, and remember.
```

> DNSLink uses [DNS TXT](https://en.wikipedia.org/wiki/TXT_record) records to map a domain name (like `ipfs.io`) to an IPFS address. Because you can edit your DNS records, you can use them to always point to the latest version of an object in IPFS (remember that an IPFS object’s address changes if you modify the object). Because DNSLink uses DNS records, the names it produces are also usually easy to type and read.

Author: Good call - incorporating.

A DNSLink address looks like an [IPNS](/guides/concepts/ipns) address, but it uses a domain name in place of a hashed public key:

```
/ipns/myexampledomain.org
```

> Just like normal IPFS addresses, they can include links to other files:

Reviewer: > Just like normal IPFS addresses, they can include links to other files:

This is supposed to be an intro to the concept of DNSLink but I think it's still worth clarifying that DNSLink addresses can link to other types of resources (e.g. directories and symlinks) that IPFS/UnivFS supports, and **not just files.**

Author: Good call - incorporating.

> Just like normal IPFS addresses, they can include links to other files:

Author: ```suggestion
Just like normal IPFS addresses, they can include links to other files — or other types of resources that IPFS/UnivFS supports, like directories and symlinks:
```

> Just like normal IPFS addresses, they can include links to other files — or other types of resources that IPFS/UnivFS supports, like directories and symlinks:

Author: ```suggestion
Just like normal IPFS addresses, they can include links to other files — or other types of resources that IPFS supports, like directories and symlinks:
```

```
/ipns/myexampledomain.org/media/
```

### Publishing using a subdomain

> While you can publish the TXT record to the exact domain (as in the example above) if you so wish, it can be more advantageous to publish DNSLink records using a special subdomain called `_dnslink`. This enables you to improve the security of an automated setup, or delegate control over your DNSLink records to a third party without giving away full control over the original DNS zone.

Reviewer: > While you can publish the TXT record to the exact domain (as in the example above) if you so wish,

There's no "example above" anymore: it was deleted in this commit.

The rest of the paragraph is cool.

Author: Thanks for catching that!

> While you can publish the TXT record to the exact domain (as in the example above) if you so wish, it can be more advantageous to publish DNSLink records using a special subdomain called `_dnslink`. This enables you to improve the security of an automated setup, or delegate control over your DNSLink records to a third party without giving away full control over the original DNS zone.

Author: ```suggestion
While you can publish the TXT record to the exact domain if you so wish, it can be more advantageous to publish DNSLink records using a special subdomain called `_dnslink`. This enables you to improve the security of an automated setup, or delegate control over your DNSLink records to a third party without giving away full control over the original DNS zone.
```

For example, [`docs.ipfs.io`](https://docs.ipfs.io) does not have a TXT record, but the page still loads
because a TXT record exists for `_dnslink.docs.ipfs.io`. If you look up the DNS records for `_dnslink.docs.ipfs.io`, you'll see its DNSLink entry:

```sh
$ dig +noall +answer TXT _dnslink.docs.ipfs.io
_dnslink.docs.ipfs.io.  34  IN  TXT "dnslink=/ipfs/QmVMxjouRQCA2QykL5Rc77DvjfaX6m8NL6RyHXRTaZ9iya"
```

### Resolving using DNSLink

> When an IPFS client or node attempts to resolve an address, it looks for a `TXT` record for the domain or subdomain in question with content like:

Reviewer: While it's true what's written, it's more correct to say that it looks for a `TXT` record that is prefixed with `dnslink=`. The rest can be an `/ipfs/` link (as in the example) or `/ipns/` or even a link to another DNSLink.

But then again, this is just an intro, and the user probably should just check dnslink.io for details.

Author: Good distinction! Incorporating.

> When an IPFS client or node attempts to resolve an address, it looks for a `TXT` record for the domain or subdomain in question with content like:

Author: ```suggestion
When an IPFS client or node attempts to resolve an address, it looks for a `TXT` record that is prefixed with `dnslink=`. The rest can be an `/ipfs/` link (as in the example below), or `/ipns/`, or even a link to another DNSLink.
```

```
dnslink=/ipfs/<CID for your content here>
```

For example, let's go back to when we looked up the DNS records for `_dnslink.docs.ipfs.io` and saw its DNSLink entry:

```sh
$ dig +noall +answer TXT _dnslink.docs.ipfs.io
_dnslink.docs.ipfs.io.  34  IN  TXT "dnslink=/ipfs/QmVMxjouRQCA2QykL5Rc77DvjfaX6m8NL6RyHXRTaZ9iya"
```

Based on that, this address:

```

> /ipns/_dnslink.docs.ipfs.io/introduction/

Reviewer 2: `_dnslink` is omitted from `/ipns/` paths:

```suggestion
/ipns/docs.ipfs.io/introduction/
```

## What is DNSLink?

> DNSLink uses [DNS TXT](https://en.wikipedia.org/wiki/TXT_record) records to map a domain name (like `ipfs.io`) to an IPFS address. Because you can edit your DNS records, you can use them to always point to the latest version of an object in IPFS (remember that an IPFS object’s address changes if you modify the object). Since DNSLink uses DNS records, you can assign names/paths/(sub)domains/whatever that are easy to type, read, and remember.

Reviewer: > names/paths/(sub)domains/whatever

Slashes are being overused (again, originally by me); also, "whatever" sounds a little too informal.

*PS: I am new to collaboration on GitHub and I'm just starting to get the hang of it. Next time, I'll try to write serious "edit suggestions" in stead of "general ideas that need to be reworded".*

Author: @dreamski21 IMO it doesn’t seem too informal at all; more like friendly. Let’s leave it in! You’ve done an awesome job on these. In fact, we’d love any suggestions on any other docs you work through along your IPFS journey. This help is greatly appreciated!

Reviewer: Happy to help. That's the least I could do, especially after the mess I made at [ipfs/go-ipfs](https://github.com/ipfs/go-ipfs/pulls/dreamski21) (trying to "make up for my sins" XD).

/ipns/myexampledomain.org
```

> Just like normal IPFS addresses, they can include links to other files — or other types of resources that IPFS/UnivFS supports, like directories and symlinks:

Reviewer: IPFS/UnixFS\* (I made that typo in my original comment XD)

Also, when I wrote "IPFS/UnixFS", it was just for clarification, I was not suggesting that you should include that exact wording (because UnixFS is more like an implementation detail) -- "**or other types of resources that IPFS supports, like directories and symlinks**" should suffice.

*Disclaimer: I'm just a curious user, not an 'information architect' or anything and I have only recently started learning about IPFS.*

Author: Fixed. Thanks again for all of your diligent efforts!
